### PR TITLE
Fixes issues while loading `word2vec` and `doc2vec` models saved using old Gensim versions. Fix #2000, #1977

### LINKED ISSUE #2000: 'Word2Vec' object has no attribute 'trainables'
```
from gensim.models import word2vec
model = word2vec.Word2Vec.load('myModel')
```

#### Description
I got this:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/gensim/models/word2vec.py", line 975, in load
    return super(Word2Vec, cls).load(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/gensim/models/base_any2vec.py", line 629, in load
    model = super(BaseWordEmbeddingsModel, cls).load(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/gensim/models/base_any2vec.py", line 278, in load
    return super(BaseAny2VecModel, cls).load(fname_or_handle, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/gensim/utils.py", line 426, in load
    obj._load_specials(fname, mmap, compress, subname)
  File "/usr/local/lib/python3.5/dist-packages/gensim/utils.py", line 469, in _load_specials
    setattr(self, attrib, val)
  File "/usr/local/lib/python3.5/dist-packages/gensim/utils.py", line 1398, in new_func1
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/gensim/models/base_any2vec.py", line 380, in syn1neg
    self.trainables.syn1neg = value
AttributeError: 'Word2Vec' object has no attribute 'trainables'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "word.py", line 164, in <module>
    X_vector, Y_vector = XY_vector(X, Y)
  File "word.py", line 116, in XY_vector
    model = gensim.models.word2vec.Word2Vec.load('./word_vector/Word60.model')
  File "/usr/local/lib/python3.5/dist-packages/gensim/models/word2vec.py", line 979, in load
    return load_old_word2vec(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/gensim/models/deprecated/word2vec.py", line 195, in load_old_word2vec
    new_model.min_alpha_yet_reached = old_model.min_alpha_yet_reached
AttributeError: 'Word2Vec' object has no attribute 'min_alpha_yet_reached'

```

#### Versions
I used gensim version 3.4.0, python version 3.5.2

#### Please share your model (very important for reproducing your error)
What should I do for this?

#### What's a python and OS?
ubuntu16.04LTS

### PR DESCRIPTION
**Pull request by manneshiva**
This PR addresses #2000 and #1977. The issues were caused due to a few missing attributes (like `min_alpha_yet_reached` , `running_training_loss`) in really old Gensim versions. I have added tests to load `word2vec` model saved using Gensim version `0.12.0` and `doc2vec` model saved using Gensim `0.13.0`. The tests also include checking online training and a similarity search post loading these old models.

### CODE REVIEW

**Comment by menshikh-iv** [comment]
Plan:
- Add toy models for each version (before 3.4.0), add tests for it too.
- Fix additional errors (if happens).

We need to cover as much as possible situations in this case.

File: gensim/test/test_word2vec.py [hunk 1]
```diff
@@ -789,6 +789,20 @@ def testLoadOldModel(self):
...
+            '0.12.0', '0.12.1', '0.12.2', '0.12.3', '0.12.4',
+            '0.13.0', '0.13.1', '0.13.2', '0.13.3', '0.13.4',
+            '1.0.0', '1.0.1', '2.0.0', '2.1.0', '2.2.0', '2.3.0',
+            '3.0.0', '3.1.0', '3.2.0', '3.3.0', '3.4.0'
+        ]
+
+        saved_models_dir = datapath('old_w2v_models')
+        for old_version in old_versions:
+            model = word2vec.Word2Vec.load(os.path.join(saved_models_dir, 'w2v_{}.mdl'.format(old_version)))
+            self.assertTrue(len(model.wv.vocab) == 3)
```
Comment (menshikh-iv) [inline gensim/test/test_word2vec.py]: add `most_similar` + update an model (similar for d2v)

File: gensim/models/deprecated/doc2vec.py [hunk 2]
```diff
@@ -138,11 +138,12 @@ def load_old_doc2vec(*args, **kwargs):
     # set doc2vec vocabulary attributes
     new_model.docvecs.doctags = old_model.docvecs.doctags
     new_model.docvecs.count = old_model.docvecs.count
-    if hasattr(old_model.docvecs, 'max_rawint'): # `doc2vec` models before `0.12.3` do not have these 2 attributes
+    if hasattr(old_model.docvecs, 'max_rawint'):  # `doc2vec` models before `0.12.3` do not have these 2 attributes
         new_model.docvecs.max_rawint = old_model.docvecs.__dict__.get('max_rawint')
         new_model.docvecs.offset2doctag = old_model.docvecs.__dict__.get('offset2doctag')
     else:
-        new_model.docvecs.max_rawint = len(old_model.docvecs.index2doctag) if old_model.docvecs.index2doctag else old_model.docvecs.count
+        new_model.docvecs.max_rawint = \
```
Comment (piskvorky) [inline gensim/models/deprecated/doc2vec.py]: Magic: definitely deserves a comment.

File: gensim/test/test_doc2vec.py [hunk 3]
```diff
@@ -122,6 +122,31 @@ def testLoadOldModel(self):
...
+            model = doc2vec.Doc2Vec.load(os.path.join(saved_models_dir, 'd2v_{}.mdl'.format(old_version)))
+            self.assertTrue(len(model.wv.vocab) == 3)
+            self.assertTrue(model.wv.vectors.shape == (3, 4))
+            self.assertTrue(model.docvecs.vectors_docs.shape == (2, 4))
+            self.assertTrue(model.docvecs.count == 2)
+            # check if inferring vectors for new documents and similarity search works.
+            doc0_inferred = model.infer_vector(list(DocsLeeCorpus())[0].words)
+            sims_to_infer = model.docvecs.most_similar([doc0_inferred], topn=len(model.docvecs))
+            self.assertTrue(sims_to_infer)
+
```
Comment (menshikh-iv) [inline gensim/test/test_doc2vec.py]: Can you add here `save`+`load`+`infer_vector` here (to be 100% sure that this persistent correctly)? Make sure that you used `/tmp` directory, check `gensim.test.utils`, you'll found needed functions (and same for w2v).

Also, please try to update model (as for w2v)

File: gensim/test/test_doc2vec.py [hunk 4]
```diff
@@ -122,6 +122,31 @@ def testLoadOldModel(self):
...
+        # Test loading doc2vec models from all previous versions
+        old_versions = [
+            '0.12.0', '0.12.1', '0.12.2', '0.12.3', '0.12.4',
+            '0.13.0', '0.13.1', '0.13.2', '0.13.3', '0.13.4',
+            '1.0.0', '1.0.1', '2.0.0', '2.1.0', '2.2.0', '2.3.0',
+            '3.0.0', '3.1.0', '3.2.0', '3.3.0', '3.4.0'
+        ]
+
+        saved_models_dir = datapath('old_d2v_models')
```
Comment (menshikh-iv) [inline gensim/test/test_doc2vec.py]: better `datapath('old_d2v_models/d2v_{}.mdl') and format later

File: gensim/test/test_word2vec.py (hunk 1 above)
Comment (menshikh-iv) [reply gensim/test/test_word2vec.py]: still missing `most_similar`

**Pull request merged**